model/test_log.go

```diff
@@ -60,14 +64,31 @@ func FindOneTestLog(name, task string, execution int) (*TestLog, error) {
 	return tl, errors.WithStack(err)
 }
 
+func findAllTestLogs(query db.Q) ([]TestLog, error) {
+	var result []TestLog
+	if err := db.FindAllQ(TestLogCollection, query, &result); err != nil {
+		return nil, errors.Wrap(err, "finding test logs")
+	}
+	return result, nil
+}
+
 func DeleteTestLogsWithLimit(ctx context.Context, env evergreen.Environment, ts time.Time, limit int) (int, error) {
-	if limit > 100*1000 {
-		panic("cannot delete more than 100k documents in a single operation")
+	if limit > maxDeleteCount {
+		return 0, errors.Errorf("cannot delete more than %d documents in a single operation", maxDeleteCount)
+	}
+
+	docsToDelete, err := findAllTestLogs(db.Query(bson.M{TestLogIdKey: bson.M{"$lt": primitive.NewObjectIDFromTimestamp(ts).Hex()}}).WithFields(TestLogIdKey).Limit(limit))
+	if err != nil {
+		return 0, errors.Wrap(err, "getting docs to delete")
+	}
+
+	if len(docsToDelete) == 0 {
+		return 0, nil
 	}
 
-	ops := make([]mongo.WriteModel, limit)
-	for idx := 0; idx < limit; idx++ {
-		ops[idx] = mongo.NewDeleteOneModel().SetFilter(bson.M{TestLogIdKey: bson.M{"$lt": primitive.NewObjectIDFromTimestamp(ts).Hex()}})
```

Comment:
This old code seems super weird... it was repeatedly "deleting one" with a range condition?

Reply:
I think the problem it was solving for was that we wanted to delete a bunch of matching docs in parallel, but with a limit. Since deleteMany doesn't take a limit parameter we used an unordered bulk write which (ostensibly) executes in parallel.